dispike/creating/models/options.py

```diff
@@ -147,10 +160,27 @@ class DiscordCommand(BaseModel):
         description (str): Description of this command.
         options (typing.List[typing.Union[SubcommandOption, CommandOption]]): Options for this command.
         default_permission (boolean): whether the command is enabled by default when the app is added to a guild. Defaults to True.
+        type (Union[CommandTypes, int]): The type of command. This defaults to SLASH
     """
 
     id: typing.Optional[int]
     name: str
-    description: str
-    options: typing.List[typing.Union[SubcommandOption, CommandOption]]
+    description: str = ""
+    options: typing.List[typing.Union[SubcommandOption, CommandOption]] = []
     default_permission: bool = True
+    type: typing.Union[CommandTypes, int] = CommandTypes.SLASH
+
+    def __init__(self, type=CommandTypes.SLASH, description="", options=[], **data) -> None:
```

Comment:
I'd rather not do this, it may break things down (gut feeling: pydantic).. Luckily pydantic has some validation features that can do this.. Here is one I wrote up 

```python

from pydantic import root_validator, validator

    @root_validator
    def validate_correct_type_requirements(cls, values):
        _cls_type, _cls_description, _cls_options = (
            values.get("type"),
            values.get("description"),
            values.get("options"),
        )
        if type == CommandTypes.SLASH:
            if _cls_description == "":
                raise ValueError("Slash commands require a description")
        if type == CommandTypes.MESSAGE or type == CommandTypes.USER:
            if _cls_description != "":
                raise ValueError("Context commands cannot have a description")
            if _cls_description:
                raise ValueError("Context commands cannot have options")

        return values

    @validator("type", pre=True, always=True)
    def set_value_type(cls, v):
        if isinstance(v, CommandTypes):
            return v.value
        return v
```

Reply:
I knew you would say that, should of left a comment. I spent the whole day reading the docs and it really didn't work no matter what I tried. So I gave in and used that instead

Reply:
Yeah no problem, it threw me off guard as well.